File: IJA/ukol2/ija/ija2019/homework2/maps/Stop.java
package ija.ija2019.homework2.maps;

import ija.ija2019.homework2.maps.*;


public interface Stop{

    public Coordinate getCoordinate();
    public String getId();
    public Street getStreet();
    public void setStreet(Street s);
    public static Stop defaultStop(String id, Coordinate c){
        Stop stop = new Stop();
        stop.id = id;
        stop.c = c;
    }



}

File: ICP/proj1/Stop.cpp
#include "Stop.h"

using namespace std;

Stop::Stop(String str, Coordinate c)
{
    Id = str;
    cords = c;
}

Stop::Stop(String str)
{
    Id = str;

}

String getId()
{

    return Id;
}

Coordinate::Coordinate getCoordinate();
{
    return cords;
}

void setStreet(Street::Street s)
{
    street = s;
}

Street::Street getStreet()
{
    return street;
}

Stop::Stop clone()
{
    Coordinate::Coordinate c2 = this.getCoordinate();
    Coordinate::Coordinate c = Coordinate(c2.getX(), c2.getY());
    return new Stop(this.getId(), c);
}
